### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by Agoni00** [NOAA-EMC/WW3#504] [Q&A]
I created a case and it runs from June 1st to December 31st, 1996. When the mode runs to September 1st, the following error occurs. What is the error?

<img width="1119" alt="微信图片_20211022193745" src="https://user-images.githubusercontent.com/92639723/138447462-e5690b36-6529-4205-a69f-8b01201da22e.png">

**Comment by JessicaMeixner-NOAA** [comment]
To get a more helpful traceback error, I'd recommend adding -g -traceback to your compiler flags.

**Comment by Agoni00** [comment]
@JessicaMeixner-NOAA 
Sorry I just started learning ww3, I don’t understand you too much,   I want to know how to add -g -traceback to my compiler flags.

**Reply by JessicaMeixner-NOAA** [reply]
@Agoni00 I have converted this do a discussion for now. 

Which version of the code are you using?  You want to change your comp file to add the -g and -traceback flags, but I need to know which code you are using to know how you generated your comp/link files to be more specific, but in general in your comp file you want to change the line

```
  # compilation options
  opt=
```
so that the string in opt also includes -g and -traceback

**Reply by Agoni00** [reply]
@JessicaMeixner-NOAA 

Thanks, I have found this line and added -g and -traceback, the version of my code is WW3-6.07.1 
Now my error is as follows: 
![b2f2b6538e1ccfe80edceef28d3dd4a](https://user-images.githubusercontent.com/92639723/138486052-29a93cc0-ee83-4ecf-b0e5-7846a0cf36b0.jpg)

**Reply by JessicaMeixner-NOAA** [reply]
Now the next step is to look in your tmp/ folder to see what line 549 of w3src2md.F90 is, and perhaps add a write statement in the corresponding ftn/w3src2md.ftn file to determine why you are getting a seg-fault.    Sometimes seg-faults are memory issues, so changing your set-up to allow for more memory can be useful.  You can also see https://github.com/NOAA-EMC/WW3/wiki/FAQs-page#how-to-debug-error-messages for more information.

**Reply by Agoni00** [reply]
@JessicaMeixner-NOAA 
 Thank you very much for your help. In addition, I have another question about the first frequency. I don't know how to set the value of the first frequency in a specific case, and the corresponding standard has not been mentioned in the user manual.
Such as:In regtests, different cases have different values. I would like to know what the first frequency value is based on and how I should set it.
For example, the first frequency is 0.04118 in the first case and 0.03679 in the second.
the first
$ -------------------------------------------------------------------- $
$ WAVEWATCH III Grid preprocessor input file $
$ -------------------------------------------------------------------- $
$ Grid name (C*30, in quotes)
$
'TEST GRID (GULF OF NOWHERE) '
$
$ Frequency increment factor and first frequency (Hz) ---------------- $
$ number of frequencies (wavenumbers) and directions, relative offset
$ of first direction in terms of the directional increment [-0.5,0.5].
$ In versions 1.18 and 2.22 of the model this value was by definiton 0,
$ it is added to mitigate the GSE for a first order scheme. Note that
$ this factor is IGNORED in the print plots in ww3_outp.
$
1.1 0.04118 25 24 0.
$

the second

$ WAVEWATCH III Grid preprocessor input file
$ ------------------------------------------
'1-D PROPAGATION MERIDIAN '
$
1.1 0.03679 3 4 0.
$

**Reply by JessicaMeixner-NOAA** [reply]
@Agoni00 you have set the first frequency in the two cases just as you described.  The value of the first frequency is up to the user to specify.

**Reply by Agoni00** [reply]
@JessicaMeixner-NOAA 
When I set the first frequency, what principles do I set it according to, or I can set it at will?  Will the first frequency set differently affect the simulation results?
